### DISCUSSION TITLE
Customized table badge colors

### DISCUSSION THREAD

**Discussion by eunicevassoa** [filamentphp/filament#2109] [Help] [answered]
Hi

I would like to know if there's a way to add custom colors to table badges besides the default `success`, `warning`, `danger` and `primary` colors. Something like this:
```php
BadgeColumn::make('status')
    ->label(__('Status'))
    ->enum([
        'new' => __('New'),
        'accepted' => __('Accepted'),
        'ongoing' => __('Ongoing'),
        'rejected' => __('Rejected'),
        'ended' => __('Ended')
    ])
        ->colors([
        'blue' => 'new',
        'success' => 'accepted',
        'warning' => 'ongoing',
        'gray' => 'ended',
        'danger' => 'rejected',
    ])
```
I modified the basic colors in the tailwind.config.js file and added other colors, but cannot use the custom ones.

**Comment by danharrin** [comment, marked as answer]
Not at the moment, since Tailwind only compiles classes that are present in the HTML markup :(

**Reply by eunicevassoa** [reply]
Thank you for the response 
That's sad... :(

**Reply by danharrin** [reply]
I think there's a "safe list" of classes you can add to the Tailwind config file, if you maybe pass the custom color classes in there, and then pass the classes to `->color()` ?

**Reply by eunicevassoa** [reply]
Thank you very much!
I was able to add custom colors  with safelist.

Added the classes that I wanted to the safelist inside `tailwind.config.js`:
```javascript
module.exports = {
...
    safelist: [
      'bg-blue-500/10',
      'text-blue-500',
    ],
...
};
```

And added the classes to the badge code:
```php
BadgeColumn::make('status')
    ->label(__('Status'))
    ->enum([
        'new' => __('New'),
        'accepted' => __('Accepted'),
        'ongoing' => __('Ongoing'),
        'rejected' => __('Rejected'),
        'ended' => __('Ended')
    ])
    ->colors([
        'bg-blue-500/10 text-blue-500' => 'new',
        'success' => 'accepted',
        'warning' => 'ongoing',
        'danger' => 'ended',
        'danger' => 'rejected',
    ])
```
